shortcodes/ucf-charts-shortcode.php

```diff
@@ -0,0 +1,80 @@
+<?php
+/**
+ * Handles registering the ucf-chart shortcode
+ **/
+if ( ! class_exists( 'UCF_Chart_Shortcode' ) ) {
+	class UCF_Chart_Shortcode {
+		/**
+		 * Registers the ucf-chart shortcode
+		 * @author Jim Barnes
+		 * @since 1.0.0
+		 **/
+		public static function register() {
+			add_shortcode( 'ucf-chart', array( 'UCF_Chart_Shortcode', 'handler' ) );
+		}
+
+		/**
+		 * The handler for the ucf-chart shortcode
+		 * @author Jim Barnes
+		 * @since 1.0.0
+		 * @param array $atts The array of shortcode attributes
+		 * @return string The shortcode markup
+		 **/
+		public static function handler( $atts ) {
+			$atts = shortcode_atts(
+				array(
+					'id'    => null,
+					'class' => ''
+				),
+				$atts
+			);
+
+			$id    = $atts['id'];
+			$class = $atts['class'];
+
+			if ( $id ) {
+				$chart = get_post( $id );
+
+				if ( 'ucf_chart' !== $chart->post_type ) return '';
```

Comment:
Consider checking to make sure get_post() actually returns something valid before trying to access the post_type property here (so that we're not attempting to access a property on a null value if get_post() failed and returned null).